spec/acceptance/init_spec.rb

```diff
@@ -10,4 +10,31 @@
       end
     end
   end
+
+  context 'with wg-quick' do
+    let :facts do
+      facts
+    end
+
+    it 'work with no errors' do
+      pp = <<-EOS
+         wireguard::interface { 'tun0':
```

Comment:
I think we can extend this snippet so that it creates multiple `wireguard::interface` with each provider so that the multiple class declaration error is triggered and we confirm it is fixed.